components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -3164,14 +3167,24 @@ public static boolean isJITProvisionEnhancedFeatureEnabled() {
      * @throws ClaimManagementException
      */
     public static List<ClaimMapping> getFilteredScopeClaims(List<String> claimListOfScopes,
-                                                            List<ClaimMapping> claimMappings)
-            throws ClaimManagementException {
+                                                            List<ClaimMapping> claimMappings, String tenantDomain)
+            throws ClaimManagementException, CarbonException, UserStoreException {
 
-        ClaimManagerHandler handler = ClaimManagerHandler.getInstance();
         List<String> claimMappingListOfScopes = new ArrayList<>();
-        for (String claim : claimListOfScopes) {
-            org.wso2.carbon.user.api.ClaimMapping currentMapping = handler.getClaimMapping(claim);
-            claimMappingListOfScopes.add(currentMapping.getClaim().getClaimUri());
+        UserRealm realm = AnonymousSessionUtil.getRealmByTenantDomain(
+                FrameworkServiceComponent.getRegistryService(),
+                FrameworkServiceComponent.getRealmService(), tenantDomain);
+        ClaimManager claimManager = realm.getClaimManager();
+        if (claimManager != null) {
+            for (String claim : claimListOfScopes) {
+                org.wso2.carbon.user.api.ClaimMapping currentMapping = claimManager.getClaimMapping(claim);
+                try {
+                    claimMappingListOfScopes.add(currentMapping.getClaim().getClaimUri());
+                } catch (NullPointerException e) {
```

Comment:
let's do a null check rather can catching the Exception

Reply:
and in the error log, add the details like, which claim is missing in which tenant.

Reply:
Addressed with commit 6f87afcaee2129b1d3f3c7f8d82209f9d5a982cf